### ISSUE TITLE
bazel 4.1.0 can't find javabase on Apple M1 macmini

### ISSUE THREAD

**Issue by devjgm** [bazelbuild/bazel#13573]
### Description of the problem / feature request:

bazel gives an error and asks for the `--host_javabase` when building my project on an Apple M1 macmini.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```console
$ git clone https://github.com/googleapis/google-cloud-cpp
...
$ USE_BAZEL_VERSION=4.1.0 bazelisk --bazelrc=/dev/null build ...
ERROR: /private/var/tmp/_bazel_jgm/e74a2c349bb49ff8069f80cc5514c617/external/bazel_tools/tools/jdk/BUILD:490:6: Configurable attribute "actual" doesn't match this configuration: Could not find a JDK for host execution environment, please explicitly provide one using `--host_javabase.`
...
```
### What operating system are you running Bazel on?

Apple M1 macmini

```console
$ sw_vers
ProductName:    macOS
ProductVersion: 11.4
BuildVersion:   20F71

$ arch
arm64
```

### What's the output of `bazel info release`?

```console
$ USE_BAZEL_VERSION=4.1.0 bazelisk info release
release 4.1.0
```
```console
$ USE_BAZEL_VERSION=4.1.0 bazelisk info
bazel-bin: /private/var/tmp/_bazel_jgm/e74a2c349bb49ff8069f80cc5514c617/execroot/com_github_googleapis_google_cloud_cpp/bazel-out/darwin_arm64-fastbuild/bin
bazel-genfiles: /private/var/tmp/_bazel_jgm/e74a2c349bb49ff8069f80cc5514c617/execroot/com_github_googleapis_google_cloud_cpp/bazel-out/darwin_arm64-fastbuild/bin
bazel-testlogs: /private/var/tmp/_bazel_jgm/e74a2c349bb49ff8069f80cc5514c617/execroot/com_github_googleapis_google_cloud_cpp/bazel-out/darwin_arm64-fastbuild/testlogs
character-encoding: file.encoding = ISO-8859-1, defaultCharset = ISO-8859-1
command_log: /private/var/tmp/_bazel_jgm/e74a2c349bb49ff8069f80cc5514c617/command.log
committed-heap-size: 154MB
execution_root: /private/var/tmp/_bazel_jgm/e74a2c349bb49ff8069f80cc5514c617/execroot/com_github_googleapis_google_cloud_cpp
gc-count: 13
gc-time: 291ms
install_base: /var/tmp/_bazel_jgm/install/d0b631b3e5a643567f221e133629d965
java-home: /private/var/tmp/_bazel_jgm/install/d0b631b3e5a643567f221e133629d965/embedded_tools/jdk
java-runtime: OpenJDK Runtime Environment (build 11.0.10+9-LTS) by Azul Systems, Inc.
java-vm: OpenJDK 64-Bit Server VM (build 11.0.10+9-LTS, mixed mode) by Azul Systems, Inc.
max-heap-size: 4294MB
output_base: /private/var/tmp/_bazel_jgm/e74a2c349bb49ff8069f80cc5514c617
output_path: /private/var/tmp/_bazel_jgm/e74a2c349bb49ff8069f80cc5514c617/execroot/com_github_googleapis_google_cloud_cpp/bazel-out
package_path: %workspace%
release: release 4.1.0
repository_cache: /var/tmp/_bazel_jgm/cache/repos/v1
server_log: /private/var/tmp/_bazel_jgm/e74a2c349bb49ff8069f80cc5514c617/java.log.appleslice.jgm.log.java.20210611-123309.95376
server_pid: 95376
used-heap-size: 72MB
workspace: /Users/jgm/google-cloud-cpp
```

###  Have you found anything relevant by searching the web?

https://github.com/bazelbuild/bazel/issues/10377 is similar but for Power9

### Any other information, logs, or outputs that you want to share?

**Adding the `--host_javabase` flag works**

```console
$ USE_BAZEL_VERSION=4.1.0 bazelisk --bazelrc=/dev/null build --host_javabase=@local_jdk//:jdk ...
... 
```

**Comment by comius** [comment]
Bazel uses remote JDKs for the host_javabase and has a preconfigured set in /src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE.tmpl

The set is missing M1 architecture, so the --host_javabase workaround is the simplest.
You can also import M1 remote JDK in your own WORKSPACE file.

**Comment by keith** [comment]
This should be fixed on head

**Comment by brentleyjones** [comment]
@keith Do you know if the required commits have been cherry-picked into the 4.2.0 RC?

**Comment by keith** [comment]
not sure, it has been too hard to follow

**Comment by PengKuang** [comment]
I still see the same error. Is it fixed? If not, why this issue is closed?

**Comment by Jdban** [comment]
Om 4.2.1 I get this error on my M1 mac still:

```Configurable attribute "actual" doesn't match this configuration: Could not find a JDK for host execution environment, please explicitly provide one using `--host_javabase.```

**Comment by wookayin** [comment]
bazel cannot find a proper JAVA base installed on arm64 macs. I had java11 installed via homebrew, so I had to add the following command line flag: `--define=ABSOLUTE_JAVABASE=/opt/homebrew/opt/java11` (Feel free to change the JAVABASE path according to your installation):

```
bazel test \
    --define=ABSOLUTE_JAVABASE=/opt/homebrew/opt/java11 \
    --host_javabase=@bazel_tools//tools/jdk:absolute_javabase \
    --host_java_toolchain=@bazel_tools//tools/jdk:toolchain_vanilla \
    --java_toolchain=@bazel_tools//tools/jdk:toolchain_vanilla \
    //...
```

Or put these lines in `.bazelrc` file:
```
build --define=ABSOLUTE_JAVABASE=/opt/homebrew/opt/java11
build --host_javabase=@bazel_tools//tools/jdk:absolute_javabase
build --host_java_toolchain=@bazel_tools//tools/jdk:toolchain_vanilla
build --java_toolchain=@bazel_tools//tools/jdk:toolchain_vanilla
```

See https://stackoverflow.com/questions/59040291/configuring-bazel-to-build-test-using-a-specific-jvm-version as well for reference.

**Comment by scoquelin** [comment]
Day 2 using `bazel` here so be nice 😉 

I was facing the same issue with bazel `4.2.2` and new M1 laptop and first used the workaround provided by @wookayin

Then while searching I noticed this PR : https://github.com/bazelbuild/platforms/pull/22

I went ahead and added this to my `WORKSPACE` file (or any relevant .bzl file) as described [here](https://github.com/bazelbuild/platforms/releases/tag/0.0.4) 

```
http_archive(
    name = "platforms",
    urls = [
        "https://mirror.bazel.build/github.com/bazelbuild/platforms/releases/download/0.0.4/platforms-0.0.4.tar.gz",
        "https://github.com/bazelbuild/platforms/releases/download/0.0.4/platforms-0.0.4.tar.gz",
    ],
    sha256 = "079945598e4b6cc075846f7fd6a9d0857c33a7afc0de868c2ccb96405225135d",
)
```

and then seems like the remote jdk gets downloaded and issue is gone on my end..
Hopefully someone will make sense of this 🤞 Hope that helps...
